src/core/lib/event_engine/thread_pool.h

```diff
@@ -88,12 +88,19 @@ class ThreadPool final : public grpc_event_engine::experimental::Forkable {
     explicit State(int reserve_threads) : queue(reserve_threads) {}
     Queue queue;
     ThreadCount thread_count;
+    // After pool creation we use this to rate limit creation of threads to one
+    // at a time.
+    std::atomic<bool> starting_thread{false};
```

Comment:
It took a few tries to interpret `starting_thread` correctly (is something now called a "starting thread"? is the thread starting? should a thread be started? ah that.). Maybe something like `should_start_new_thread` would be clearer.

Reply:
chose a new name